mne/io/eeglab/eeglab.py

```diff
@@ -55,34 +55,48 @@ def _to_loc(ll):
     if isinstance(ll, (int, float)) or len(ll) > 0:
         return ll
     else:
-        return 0.
+        return np.nan
 
 
 def _get_info(eeg, montage, eog=()):
     """Get measurement info."""
+    from scipy import io
     info = _empty_info(sfreq=eeg.srate)
+    update_ch_names = True
 
     # add the ch_names and info['chs'][idx]['loc']
     path = None
     if not isinstance(eeg.chanlocs, np.ndarray) and eeg.nbchan == 1:
             eeg.chanlocs = [eeg.chanlocs]
 
     if len(eeg.chanlocs) > 0:
-        ch_names, pos = list(), list()
+        pos_fields = ['X', 'Y', 'Z']
+        if (isinstance(eeg.chanlocs, np.ndarray) and not isinstance(
+                eeg.chanlocs[0], io.matlab.mio5_params.mat_struct)):
+            has_pos = all([fld in eeg.chanlocs[0].dtype.names
+                           for fld in pos_fields])
+        else:
+            has_pos = all([hasattr(eeg.chanlocs[0], fld)
+                           for fld in pos_fields])
```

Comment:
does not need `[]` in fact it's better not to have it as it will  short-circuit if possible